### DISCUSSION TITLE
range example please

### DISCUSSION THREAD

**Discussion by cekvenich2** [kriszyp/lmdb-js#105] [Q&A] [answered]
Hi, I'm trying to use range to 'scan' rows.

I'm confused by start/end args. Is there a real example? 
It is not same as offset.

tia for past support.

**Comment by kriszyp** [comment]
There are quite a few examples of range arguments in the tests, like https://github.com/DoctorEvidence/lmdb-js/blob/master/test/index.test.js#L396, although not sure if there is something more real or specific you are looking for. The `start` and `end` properties specify the key range, and then within that range, the `offset` specifies the number of entries to skip before returning/iterating entries, and the `limit` specifies the maximum number of entries to return/iterate through. Are you looking for a specific use case?

**Comment by cekvenich2** [comment]
I'm looking for hello world.
Is 1 and 100 good for start and end?

**Reply by kriszyp** [reply]
Ok, here is a simple example:
```
let db = open('my-db');
for (let i = 0; i < 50; i++) {
  // adds 50 entries, with keys from 100 - 149
  await db.put(i + 100, 'value' + i);
}
for (let { key, value } of getRange({ start: 120, end: 130 }) {
  // this would iterate entries from 120 - 129
}
for (let { key, value } of getRange({ start: 200 }) {
  // there are no keys over 200, so this wouldn't find any entries
}
for (let { key, value } of getRange({ start: 100, end: 130, offset: 10 }) {
  // this would start at 100, skip 10 entries, and iterate entries 110 - 129
}
```

**Comment by cekvenich2** [comment]
@kriszyp THANK YOU.
So you assume some kind of a sequence # for key. If I used a random GUID, then one could not get range. There would be no way 'scan'.
Based on your example, and docs - that is my understanding now. Correct?

I could use db.getKeys to scan from start, and use limit and offset to get 1K rows at a time and continue.

**Comment by kriszyp** [comment, marked as answer]
Yes, that's correct, there wouldn't be anything meaningful about doing a range query on random GUIDs. But, like you said, you could still do range query of the whole database, and use offset/limit to get sections (and you could do that by just omitting the start and end properties, and then the range will be across the whole database):
```
let iterable = db.getRange({
  offset: 2000, // start at row 2000
  limit: 1000, // iterate through a thousand
})
```

**Comment by cekvenich2** [comment]
Got it thanks.
